File: templates/hexagonal-spring-boot/domain-api/src/main/java/packageName/domain/port/ObtainExample.java
package packageName.domain.port;

import java.util.List;
import packageName.domain.model.Example;

public interface ObtainExample {

  default List<Example> getAllExamples() {
    Example example = Example.builder().id(1L).description("If you could read a leaf or tree\r\nyoud have no need of books.\r\n-- Alistair Cockburn (1987)").build();
    return List.of(example);
  }
}

File: templates/hexagonal-spring-boot-kotlin/domain-api/src/main/kotlin/packageName/domain/port/ObtainExample.kt
package packageName.domain.port

import packageName.domain.model.Example

interface ObtainExample {

    fun getAllExamples(): List<Example> {
        val example = Example(1L, "If you could read a leaf or tree\r\nyoud have no need of books.\r\n-- Alistair Cockburn (1987)")
        return listOf(example)
    }
}
